cyences_app_for_splunk/bin/cyences_product_manager.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+import sys
+
+from splunklib.searchcommands import (
+    dispatch,
+    GeneratingCommand,
+    Configuration,
+    Option,
+)
+from cs_product_list import PRODUCTS
+
+import copy
+import logging
+import logger_manager
+
+logger = logger_manager.setup_logging("cyences_product_manager", logging.INFO)
+
+
+@Configuration()
+class CyencesProductManager(GeneratingCommand):
+    operation = Option(name="operation", require=True, default="getproducts")
+    product_name = Option(name="product_name", require=False, default="")
+
+    def validate_inputs(self):
+        if self.operation not in ["getproducts", "buildproductspecificsearches"]:
+            raise Exception(
+                "operation - allowed values: getproducts, buildproductspecificsearches"
+            )
+
+    def generate(self):
+        self.validate_inputs()
+
+        try:
+            all_products = copy.deepcopy(PRODUCTS)
+
+            # list down the product names to use in data reviewer dashboard filter
+            if self.operation == "getproducts":
+                for product in all_products:
+                    yield {"products": product["name"]}
+            # create product specific searches used in data reviewer dashboard
+            elif self.operation == "buildproductspecificsearches":
+                host_reviewer_search = ""
+                sources_reviewer_search = ""
+                metadata_count_search = "| tstats count where index=* "
```

Comment:
Shall we also do metadata_count_search also in cs_product_list.py

Reply:
metadata_count_search = "| tstats count where index=* " ==> this is one time prefix.
the **repetitive part** is the source and sourcetype filter. we can add this to the cs_product_list but in that case we have to add product specific condition in custom command for the non tstats search (because it will not have the same prefix -> line no. 45).